tests/unit/test_json_schema_component_generator.py

```diff
@@ -0,0 +1,381 @@
+                    "Nanostring CosMx SMI Experiment",
+                    "NanoString GeoMx DSP Spatial Transcriptomics Level 1",
+                    "NanoString GeoMx DSP Spatial Transcriptomics Level 3",
+                    "Other Assay",
+                    "RPPA Level 2",
+                    "RPPA Level 3",
+                    "RPPA Level 4",
+                    "SRRS Imaging Level 2",
+                    "Slide-seq Level 1",
+                    "Slide-seq Level 2",
+                    "Slide-seq Level 3",
+                    "scATAC-seq Level 1",
+                    "scATAC-seq Level 2",
+                    "scATAC-seq Level 3",
+                    "scATAC-seq Level 4",
+                    "scDNA-seq Level 1",
+                    "scDNA-seq Level 2",
+                    "scRNA-seq Level 1",
+                    "scRNA-seq Level 2",
+                    "scRNA-seq Level 3",
+                    "scRNA-seq Level 4",
+                    "scmC-seq Level 1",
+                    "scmC-seq Level 2",
+                    "Accessory Manifest",
+                    "Acute Lymphoblastic Leukemia Tier 3",
+                    "Biospecimen",
+                    "Breast Cancer Tier 3",
+                    "Clinical Data Tier 2",
+                    "Colorectal Cancer Tier 3",
+                    "Demographics",
+                    "Diagnosis",
+                    "Exposure",
+                    "Family History",
+                    "Follow Up",
+                    "Lung Cancer Tier 3",
+                    "Melanoma Tier 3",
+                    "Molecular Test",
+                    "NanoString GeoMx DSP ROI DCC Segment Annotation Metadata",
+                    "NanoString GeoMx DSP ROI RCC Segment Annotation Metadata",
+                    "Neuroblastoma and Glioma Tier 3",
+                    "Ovarian Cancer Tier 3",
+                    "Pancreatic Cancer Tier 3",
+                    "Participant Vital Status Update",
+                    "Precancer Diagnosis",
+                    "Prostate Cancer Tier 3",
+                    "Publication Manifest",
+                    "SRRS Biospecimen",
+                    "SRRS Clinical Data Tier 2",
+                    "Sarcoma Tier 3",
+                    "Therapy",
+                ],
+            ),
+        ],
+        ids=[
+            "example model - local",
+            "HTAN model - url",
+        ],
+    )
+    def test_gather_components(
+        self, helpers, data_model_location: str, expected_components: list[str]
+    ):
+        # GIVEN a data model from a local file or a url
+        if data_model_location.startswith("example"):
+            data_model_location = helpers.get_data_path(data_model_location)
+
+        # WHEN an instance of the JsonSchemaGeneratorDirector class is created
+        generator = JsonSchemaGeneratorDirector(data_model_location=data_model_location)
+        # AND the _extract_components method is called
```

Comment:
this comment can be removed? the name of the function is outdated after your recent commits.